app/controllers/users_controller.rb

```diff
@@ -0,0 +1,44 @@
+class UsersController < ApplicationController
+  before_action :authorize_request, except: :create
+  before_action :find_user, except: %i[create index]
+  def index
+    @users = User.all
+    render json: @users, status: :ok
+  end
+
+  def show
+    render json: @user, status: :ok
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      render json: @user, status: :created
+    else
+      render json: { errors: @user.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    unless @user.update(user_params)
+      render json: { errors: @user.errors.full_messages },
+             status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @user.destroy
+  end
+
+  private
+
+  def find_user
+    @user = User.find_by_username!(params[:_username])
+    rescue ActiveRecord::RecordNotFound
+      render json: { errors: 'User not found' }, status: :not_found
```

Comment:
is there a way this could be factored into the `not found` method you have in the application controller so as to make it DRY?